ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilderV1.java

```diff
@@ -300,9 +302,10 @@ private <K> KStream<K, GenericRow> buildKStream(
         .stream(streamSource.getTopicName(), consumed);
 
     final int pseudoColumnVersion = streamSource.getPseudoColumnVersion();
-    return stream
-        .transformValues(new AddKeyAndPseudoColumns<>(
-            keyGenerator, pseudoColumnVersion, streamSource.getSourceSchema().headers()));
+    stream.peek((k, v) -> { });
+    return stream.processValues(() -> new AddKeyAndPseudoColumnsProcessor<>(
+            keyGenerator, pseudoColumnVersion, streamSource.getSourceSchema().headers()),
+        Named.as("KSTREAM-TRANSFORMVALUES-0000000001"));
```

Comment:
Why is this node named explicitly?

Reply:
I have done as per test case requirement. I have asked kstream team if this looks okay. It was generating `KSTREAM-PROCESSVALUES-0000000001`

Reply:
`Processor` name by themself should actually not be critical if they change. We need to worry about topic and state store names though.

What break if the processor name changes? If it's really only the name, it should be ok to accept the change and update the tests. If it has side-effects, setting the name here would sound rights.

The other question is, if we name explicitly, could it cause side-effect with regard to auto-naming of downstream processor -- if we name expliclity, we might not us the next index, and create different name later, what could also lead to undesried side effects.